compiler/src/main/java/com/squareup/anvil/compiler/codegen/KotlinPoet.kt

```diff
@@ -62,13 +62,15 @@ internal fun ClassDescriptor.asClassName(): ClassName =
   )
 
 internal fun FqName.asClassName(module: ModuleDescriptor): ClassName {
-  try {
+  val segments = pathSegments().map { it.asString() }
+
+  // if the first sentence case is not the last segment it gets ambiguous:
+  // com.Foo.Bar could be a inner class Bar or an unconventional package com.Foo
```

Comment:
Comments are full sentences with an uppercase letter at the beginning and a dot at the end.